src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/infrastructure/config/HomeOfficeProperties.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.hmcts.reform.iahomeofficeintegrationapi.infrastructure.config;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+@Component
+@EnableConfigurationProperties
+@ConfigurationProperties("homeoffice.params")
+public class HomeOfficeProperties {
+
+    private Map<String, LookupReferenceData> homeOfficeReferenceData = new HashMap<>();
+
+    public Map<String, LookupReferenceData> getHomeOfficeReferenceData() {
+        return homeOfficeReferenceData;
+    }
+
+    public static class LookupReferenceData {
+        private String code;
+        private String description;
+
+        public String getCode() {
+            return code;
+        }
+
+        public void setCode(String code) {
```

Comment:
Class should be immutable so please remove the setters.  Spring can use a no-args constructure or you can provide annotations to create objects.

Reply:
All the other model objects are immutable. This is for feign client to fill codes from application.yml required for request

Reply:
Don't really understand.  Can you explain please?  Can you not just use a constructor?

Reply:
We have structures in the instruct request, which will always have constant values. I am loading them thru application.yml and the object is created by using @EnableConfigurationProperties
@ConfigurationProperties("homeoffice.params")
This creates a map with key as the Jsonproperty name and value as the Structure I need. 
It did not work without getters and setters
"consumer": {
            "code": "HMCTS",
            "description": "HM Courts and Tribunal Service"
        }

Reply:
I also checked with payments-api and setters were put in place only for loading feign values thru application.yml